crates/oxc_parser/src/cursor.rs

```diff
@@ -384,7 +342,13 @@ impl<'a> ParserImpl<'a> {
             if first {
                 first = false;
             } else {
-                if !trailing_separator && self.at(separator) && self.peek_at(close) {
+                if !trailing_separator
+                    && self.at(separator)
+                    && self.lookahead(|p| {
+                        p.bump_any();
+                        p.at(close)
+                    })
+                {
```

Comment:
I suggest splitting this change out into a separate PR. Presumably these lines is where the parser perf regression is coming from. Removing the `VecDeque` from the lexer should be a gain in both lexer and parser benchmarks.

Then in this block of code, I think we can refactor to avoid using lookahead at all. I think that should be possible since we're immediately bumping on to next token anyway in the next line `self.expect(separator)`. If we can do that, hopefully it'll remove the perf regression.

But it'd be useful to benchmark that in isolation without the change of removing the `VecDeque` also in the mix - hence why I think it'd be better in a separate PR.